### ISSUE TITLE
Thrust from maximum thrust and thrust coefficient

### ISSUE THREAD

**Issue by DominicDirkx** [Tudat/tudat#106]
In many applications, the use a thrust coefficients is typical for a (simplified) representation of the thrust magnitude, based on the current maximum thrust value.

**Comment by DominicDirkx** [comment]
A first step of these modifications have been added in the following branch

https://github.com/DominicDirkx/tudat/tree/SpacePlaneModifications

A unit test is included for a single set of dependencies of Isp and thrust. However, the unit tests should be extended with:

- Unit test for the use of throttle setting (maximum_thrust_multiplier)
- Unit test for the use of one and more user-defined (guidance) input variables to the guidance
- Unit test for 1 and 3 independent variables
- Unit test for each of the available dependencies of the thrust (only Mach number and dynamic pressure tested; altitude, density and pressure are not)

Also, the coefficient file reader (see https://github.com/Tudat/tudat/issues/104) should be interfaced with the ParameterizedThrustMagnitudeSettings class to allow tables with (maximum) thrust/specific impulse to be read from files.

Finally, as the functionality encapsulated in this feature is quite expansive and flexible, an example application would be very useful.

**Comment by DominicDirkx** [comment]
The branch has been updated with some modifications. We decided to not allow multiple user-defined guidance parameters for now (a warning will be shown that this feature is not tested if it is used).

An indirect test of the 3-dimensional file-reading is now include through the test of control surface deflections (1-dimensional file reading remains untested). Remaining issues are:

- Unit test for 1 independent variable
- Unit test for each of the available dependencies of the thrust (only Mach number and dynamic pressure tested; altitude, density and pressure are not)
